### PR TITLE
AlertDialog.actionsOverflowButtonSpacing and ButtonBar.overflowButtonSpacing

### PR DESCRIPTION
**Pull request by shihaohong**
## Description

This change adds button spacing in between a `ButtonBar`'s children when they overflow and are stacked in a column. 

## Related Issues

n/a

## Tests

I added the following tests:

Tests to ensure that additional space is added in between each button for ButtonBar's children and AlertDialog actions in an overflow case.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/dialog.dart [hunk 1]
```diff
@@ -20,6 +20,7 @@ import 'ink_well.dart';
 import 'material.dart';
 import 'material_localizations.dart';
 import 'theme.dart';
+import 'theme_data.dart';
```
Comment (shihaohong) [inline packages/flutter/lib/src/material/dialog.dart]: This import allows devs to jump to the definition of `MaterialTapTargetSize`.

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/test/material/dialog_test.dart [hunk 2]
```diff
@@ -563,7 +563,74 @@ void main() {
...
+        RaisedButton(
+          key: key2,
+          onPressed: () {},
+          child: const Text('Looooooooooooooong button 2'),
+        ),
+      ],
+    );
+
+    await tester.pumpWidget(
+      _appWithAlertDialog(tester, dialog),
```
Comment (gspencergoog) [inline packages/flutter/test/material/dialog_test.dart]: @shihaohong looks like this line broke the build.

Comment (gspencergoog) [reply packages/flutter/test/material/dialog_test.dart]: https://cirrus-ci.com/task/5286501055463424

Comment (shihaohong) [reply packages/flutter/test/material/dialog_test.dart]: I think another PR was merged that changed `_appWithAlertDialog`'s function signature recently. Should be an easy fix!

**Pull request merged**